packages/valory/skills/decision_maker_abci/behaviours/reedem.py

```diff
@@ -435,10 +435,9 @@ def _check_already_redeemed(self) -> WaitableConditionType:
             if not result:
                 n_retries += 1
                 keep_fraction = (1 - self.params.reduce_factor) ** n_retries
-                step = int(batch_size * keep_fraction)
-                msg = f"Repeating this call with a decreased batch size of {step}."
+                batch_size = int(self.params.event_filtering_batch_size * keep_fraction)
+                msg = f"Repeating this call with a decreased batch size of {batch_size}."
                 self.context.logger.warning(msg)
-                from_block += step
```

Comment:
This would modify the starting point of the failed query and keep the batch size unchanged.